### PR TITLE
Making manager more mobile friendly

### PR DESCRIPTION
**Pull request by jpdevries**
### What does it do ?
- Gets the fixed and absolute positioned silliness out of the way on
  small screens by combatting ExtJS' nature of aggressively preventing usability on small viewports
- fights :fire: with :fire: 
- Makes the left hand tree more responsive (can expand and collapse with a button at the bottom of the page)
- Make the resource edit screen mobile friendly
### Why is it needed ?

So it is more feasible to navigate the manager and edit resources on mobile devices.
### Related issue(s)/PR(s)

n/a

### CODE REVIEW

**Comment by OptimusCrime** [comment]
Pretty sure you should not include the rendered .css file. Other than that, awesome work!

**Comment by pixelchutes** [comment]
This looks awesome, JP!

File: manager/assets/modext/core/modx.component.js [hunk 1]
```diff
@@ -7,8 +7,9 @@ MODx.Component = function(config) {
     if (this.config.tabs) {
         this._loadTabs();
     }
-    this._loadComponents();
     this._loadActionButtons();
+	console.log(document.getElementById('modx-container').innerHTML);
```
Comment (pixelchutes) [inline manager/assets/modext/core/modx.component.js]: Any reason this shouldn't be removed?

Comment (jpdevries) [inline manager/assets/modext/core/modx.component.js]: Lol whoops! That's what I get for submitting a pr during a MODXpo lecture. I'll fix it. 



—

**Comment by jpdevries** [comment]
@OptimusCrime AFAIK it is part of the workflow to build the files before committing. When merges happen with CSS Integrators can just re-run `grunt build` to solve them.

I removed the console.log statement and cleaned up by squashing the commits into one

**Comment by Mark-H** [comment]
@jpdevries Actually @OptimusCrime is right - to prevent conflicts from occurring it's better if the compiled CSS is not included in pull requests, the integrators will take care of updating that as things get merged. It's okay though, can sort it out either way. 

Could you update your PR with the template from https://github.com/modxcms/revolution/blob/2.x/CONTRIBUTING.md#template-1? Gotta set the right example ;)

**Comment by jpdevries** [comment]
@Mark-H Oh I see, wasn't aware of that. done and done, i pulled the css out too

**Comment by Mark-H** [comment]
Thanks man :) This looks really good from your screenshot, but it looks like it might need some further tweaks. Let's look at this together tomorrow? 

screen shot 2015-11-22 at 01 08 48

**Comment by jpdevries** [comment]
Huh, ya i am not sure what is going on there. I originally branched off develop, but submitted the pr to 2.4.x. Not sure if that could have anything to do with it or not. 



—

**Comment by Mark-H** [comment]
That could be related for sure, let's find out tomorrow, could also be an issue with my dev environment as I just started a fresh one.

**Comment by jpdevries** [comment]
I reproduced it kind of but then after I cleared cache and refreshed the page it looks right again. Ya let's look at it tomorrow. It would be nice to try and figure out a way to format the uber bar too and maybe even re order the preview button (using flex-order) so it is last and big like the help button is now. 



—

**Comment by mindeffects** [comment]
Great idea for a PR, @jpdevries!
But why not kill those vertical lines in the menu bar at all? They are not needed for anything and are already gone on the right side of the manu bar.

**Comment by jpdevries** [comment]
@mindeffects i'm not opposed to removing the lines, just mostly been focusing on responsive layout with this PR.

@Mark-H I think I know what is going on. The layout is still messed up if the tree is open

### Tree Open :cry: 


### Tree Closed :smile:

**Comment by jpdevries** [comment]
With a bit more work I managed to get the lefthand sidebar (tree) to be more supported on mobile.



It is kind of weird how you expand and collapse the tree, but I'm making do with what I can. If you scroll down to the bottom of the page there is a button to expand / collapse the tree



Haven't pushed yet but will soon.

**Comment by exside** [comment]
:+1:

**Comment by jpdevries** [comment]
I intended to only work on the header (hints the responsive-header branch name) but figured I'd try and work my way down the page. How about making the content entry area a bit more mobile friendly?

**Comment by mindeffects** [comment]
Cool!

**Comment by jpdevries** [comment]
Just updated the expand / collapse sidebar button to be responsive. 

It’s a little weird because the component is all the way at the bottom
of the page. Ideally it would be just beneath the tree sidebar itself,
but we haven’t been able to figure out how to change where the
component is ordered within the markup so this is the best we can do
for now

File: manager/assets/modext/core/modx.component.js [hunk 2]
```diff
@@ -104,7 +104,7 @@ Ext.extend(MODx.Component,Ext.Component,{
         return true;
     }
 });
-Ext.reg('modx-component',MODx.Component);
```
Comment (Mark-H) [inline manager/assets/modext/core/modx.component.js]: Is there any particular reason you moved this line down to line 359?

Comment (jpdevries) [inline manager/assets/modext/core/modx.component.js]: that is from when I was trying to figure out how to change the order of where the action buttons where on the page, but I do't think all those changes to modx.component.js are needed. I'll try removing some of them

**Comment by jpdevries** [comment]
This PR does change where the MODX action bars get rendered to (so they can be before the main area) so I suppose that could cause some edge cases but all it is really doing is giving it another wrapper. If people are selecting the actions box by the id to do things like add their own buttons all that should work fine. 
Either way, I suppose targeting 2.5 is good to be safe.

**Comment by jpdevries** [comment]
I dld a little more work while I was :airplane: across the pond :smiley: 

### Uberbar Open
screen shot 2015-11-23 at 7 05 16 pm

### Create/Edit Template Variable
screen shot 2015-11-23 at 2 08 41 pm

### Create/Edit Template
screen shot 2015-11-23 at 2 08 27 pm

### Create/Edit Chunk
screen shot 2015-11-23 at 2 09 06 pm

### Properties Tab
screen shot 2015-11-23 at 3 22 32 pm

### Create New Snippet
screen shot 2015-11-23 at 3 25 04 pm

### Create New Plugin
screen shot 2015-11-23 at 3 26 07 pm

### New Category Modal
screen shot 2015-11-23 at 3 37 54 pm

### Import HTML
screen shot 2015-11-23 at 4 00 46 pm

### Media Sources
screen shot 2015-11-23 at 4 34 07 pm

### Package Management
screen shot 2015-11-23 at 4 45 28 pm

### Clear Console
screen shot 2015-11-23 at 5 02 48 pm

### Settings Page
screen shot 2015-11-23 at 7 09 31 pm

### Context Page
screen shot 2015-11-23 at 7 25 42 pm

### ACLs Page
screen shot 2015-11-23 at 7 46 29 pm

### Create Property Set
screen shot 2015-11-23 at 7 51 35 pm

### Lexicons Page
screen shot 2015-11-23 at 7 52 03 pm

### Namespaces Page
screen shot 2015-11-23 at 7 53 12 pm

**Comment by hansek** [comment]
Good job :+1:

**Comment by christianseel** [comment]
Fantastic job JP! :tada: :+1:

**Comment by mindeffects** [comment]
Awesome!
I hope this PR makes it into 2.4.3!

**Comment by jpdevries** [comment]
> I hope this PR makes it into 2.4.3!

@mindeffects I did as well but it has been slated for 2.5.0 mostly because it moves where the actions toolbar is in the DOM. It still uses the same exact id, so I see no reason why this would break anything. 

I intend to install 2.4.3 built from my fork so I can start my clients on it early. I can share the zip ;)

**Comment by eighthday** [comment]
amazing! never thought this would come before modx 3

**Comment by jaygilmore** [comment]
Squee!

**Comment by rtripault** [comment]
Absolutely awesome work & effort! Huge :+1: from here!

My only concern is about code comments (as in many other core code places, unfortunately).
SASS allows comments, which could be "removed" when compiling, so i would love having loads of those.

It's hard for other maintainers/contributors to understand the intention, or even what's supposed to be targeted, without comments.

**Comment by jpdevries** [comment]
@rtripault glad you like it!

The Sass is certainly nasty. Overly specific selectors and !important declarations are needed to override the inline styling of ExtJS. I could certainly add some comments to help organize things. Also, I would like to adjust the styles for certain components but would need some help from an ExtJS contributor that could help add a few classes in certain areas and maybe change the order of a couple of things in the DOM if possible.

**Comment by pixelchutes** [comment]
@jpdevries I haven't had a chance to test this yet, but one observation on the "Uberbar Focused" and "Uberbar Results" images above:

It's seems to have regressed the 'up arrow' visual connector, which previously was designed to align with the magnifying glass at each width / breakpoint:

### Before
*Notice at the end (once resized smaller)*
searchbar

**Comment by HarveyEV** [comment]
Awesomesauce! Can't wait for 2.5. :thumbsup:

**Comment by jpdevries** [comment]
@pixelchutes good call on the arrow that should be an easy fix

**Comment by jpdevries** [comment]
updated

**Comment by Mark-H** [comment]
[This has been merged into 2.x](https://github.com/modxcms/revolution/commit/3824267f35ae40dd07260f33c6cf67c7e8598a76) for inclusion in 2.5, thank you @jpdevries for making MODX more usable!!

**Comment by jpdevries** [comment]
@Mark-H what can I say i can't :zzz: on :airplane: 
Thanks for the merge and Happy December :smile:

**Comment by noahlearner** [comment]
Will this work with touch?  A big part of how the manager works depends on right clicking menu items.  How will we do this on mobile device / tablet?

**Comment by Mark-H** [comment]
@noahlearner This pull request doesn't change that, this just makes sure that you're actually able of navigating through the manager without wanting to poke your eyes out. :) Various mobile devices/tablets have different ways of doing right clicks.. long or force tabs typically. 

I'm sure there'll be further improvements in the future specifically to target the right clicks, but this one is not that yet.

**Comment by noahlearner** [comment]
Will the solution look like a javascript solution to listen for a hold down to map to right click?

Kind of like the solution on: http://stackoverflow.com/questions/2625210/long-press-in-javascript

or the http://www.jquery-plugins.info/longclick-00014475.htm

**Comment by rtripault** [comment]
Just rising an "slightly boring" issue : 

* change your browser viewport to a "mobile" one, ie. 615px (the tree is now at the top and its width is ~600px)
* restore to a "desktop width", notice the tree is still around 600px, with an horizontal scroollbar at the bottom

**Pull request closed without merging**